lib/Target/TfheRustBool/TfheRustBoolEmitter.cpp

```diff
@@ -215,9 +245,25 @@ void TfheRustBoolEmitter::emitReferenceConversion(Value value) {
 LogicalResult TfheRustBoolEmitter::printSksMethod(
     ::mlir::Value result, ::mlir::Value sks, ::mlir::ValueRange nonSksOperands,
     std::string_view op, SmallVector<std::string> operandTypes) {
-  if (isa<TensorType>(nonSksOperands[0].getType())) {
+  // ToDo: Check again the enum definition
+  if (packedAPI && (op == "and" || op == "nand" || op == "or" || op == "nor" ||
+                    op == "xor" || op == "xnor")) {
+    std::string gateStr = std::string(op);
+    std::transform(
+        gateStr.begin(), gateStr.end(), gateStr.begin(),
+        [](unsigned char c) -> unsigned char { return std::toupper(c); });
```

Comment:
It might be possible to use `llvm::StringRef`'s [upper()](https://github.com/llvm/llvm-project/blob/5b3ba438dfa7815bb0f3be07a300866085a431b9/llvm/lib/Support/StringRef.cpp#L116)

```suggestion
    std::string gateStr = StringRef(op).upper();
```